### PR TITLE
refactor gdoc and gsearch with new webapp architecture

### PR DESCRIPTION
**Pull request by ShakoHo**
@askeing @MikeLien @ypwalter r?

### CODE REVIEW

File: lib/sikuli/gdoc.sikuli/gdoc.py [hunk 1]
```diff
@@ -85,15 +159,18 @@ def page_text_generate(self, keyword, page):
...
-        click(Pattern("pics/FindReplaceInput.png").similar(0.5).targetOffset(98, 26))
+        self._click(action_name='Click replace button', similarity=0.5,
+                    component=gDoc.GDOC_CLICK_REPLACE_FINDANDPLACE_INPUT)
         type(replace_keyword)
         for i in range(replace_times):
-            wait(Pattern("pics/Replace.png").similar(0.8))
-            click(Pattern("pics/Replace.png").similar(0.8))
+            self._wait_for_loaded(component=gDoc.GDOC_REPLACE_ICON, similarity=0.8)
+            self._click(action_name='Click replace button', similarity=0.8,
+                    component=gDoc.GDOC_REPLACE_ICON)
```
Comment (ypwalter) [inline lib/sikuli/gdoc.sikuli/gdoc.py]: four more spaces

File: lib/sikuli/common.sikuli/common.py [hunk 2]
```diff
@@ -331,3 +339,72 @@ def il_hover(self, action_name, component, width, height, similarity=0.70, timeo
...
+        @param component: The component you want to click. ex: GMAIL_REPLY.
+        @param similarity: The similarity of image. Default: 0.70.
+        @return: true/false
+        """
+        # get the loop time base on the pattern amount of component, min loop time is 10 times
+        wait_sec = 0.5
+        loop_time = self._get_loop_times(object_amount=len(component), total_second=timeout, each_check_second=wait_sec)
+        for counter in range(loop_time):
+            for pic, offset_x, offset_y in component:
+                p = Pattern(pic).similar(similarity).targetOffset(offset_x, offset_y)
```
Comment (ypwalter) [inline lib/sikuli/common.sikuli/common.py]: why do we need `targetOffset()` for exists?

File: lib/sikuli/common.sikuli/common.py [hunk 3]
```diff
@@ -9,7 +9,11 @@ def __init__(self):
 
         if self.os.lower() == 'mac':
             self.control = Key.CMD
+            self.input_wheel_down_direction = WHEEL_UP
```
Comment (ypwalter) [inline lib/sikuli/common.sikuli/common.py]: I don't see this get used elsewhere in General class.

File: lib/sikuli/common.sikuli/common.py (hunk 2 above)
Comment (ShakoHo) [inline lib/sikuli/common.sikuli/common.py]: fulfill the pattern of the image, I will leave the interface here, so if user gives the parameter 0,0, that should not affect the result?

File: lib/sikuli/common.sikuli/common.py (hunk 3 above)
Comment (ShakoHo) [inline lib/sikuli/common.sikuli/common.py]: for future use, in case user implements same logic again..

**Pull request merged**